presto-main/src/test/java/com/facebook/presto/operator/TestPrecisionRecallAggregation.java

```diff
@@ -0,0 +1,284 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.metadata.FunctionManager;
+import com.facebook.presto.metadata.MetadataManager;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+
+import static com.facebook.presto.block.BlockAssertions.createBooleansBlock;
+import static com.facebook.presto.block.BlockAssertions.createDoublesBlock;
+import static com.facebook.presto.block.BlockAssertions.createLongsBlock;
+import static com.facebook.presto.operator.aggregation.AggregationTestUtils.assertAggregation;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.sql.analyzer.TypeSignatureProvider.fromTypes;
+import static org.testng.Assert.assertTrue;
+
+public abstract class TestPrecisionRecallAggregation
+        extends AbstractTestAggregationFunction
+{
+    private static final Integer NUM_BINS = 3;
+    private static final double MIN_FALSE_PRED = 0.2;
+    private static final double MAX_FALSE_PRED = 0.5;
+
+    private final String functionName;
+    private InternalAggregationFunction precisionRecallFunction;
+
+    @BeforeClass
+    public void setUp()
+    {
+        FunctionManager functionManager = MetadataManager.createTestMetadataManager().getFunctionManager();
+        precisionRecallFunction = functionManager.getAggregateFunctionImplementation(
+                functionManager.lookupFunction(
+                        this.functionName,
+                        fromTypes(BIGINT, BOOLEAN, DOUBLE, DOUBLE)));
+    }
+
+    @Test
+    public void testNegativeWeight()
+    {
+        String error = "";
+        try {
```

Comment:
I think we can omit the check on the specific text of the exception and wrap the `assertAggregation` with an `assertThrows(PrestoException.class, () -> assertAggregation(...))`

Reply:
Not sure I follow. If this case happens, I'd like an exception thrown which contains this text, as it is indicative of the problem that actually happened. This is what the test does.

Reply:
IMO it's sufficient to just test that it throws.  If we really want to check on the exception text, then I'd do it differently:

```java
        try {
            assertAggregation(
                    precisionRecallFunction,
                    0.0,
                    createLongsBlock(Long.valueOf(200)),
                    createBooleansBlock(Boolean.valueOf(true)),
                    createDoublesBlock(Double.valueOf(0.2)),
                    createDoublesBlock(Double.valueOf(-0.2)));
            fail("Expected PrestoException");
        }
        catch (PrestoException e) {
            assertTrue(e.getText().toLowerCase(Locale.ENGLISH).contains("weight"));
            assertTrue(e.getText().toLowerCase(Locale.ENGLISH).contains("negative"));
        }
```